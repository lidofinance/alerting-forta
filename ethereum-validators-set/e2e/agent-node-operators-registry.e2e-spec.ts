import { configureContainer, Finding } from "forta-agent";
import { AwilixContainer, asFunction } from "awilix";
import {
  provideAgentPath,
  provideRunBlock,
  provideRunTransaction,
} from "./utils";

const TEST_TIMEOUT = 60_000; // ms

describe("agent-node-operators-registry e2e tests", () => {
  let runBlock: (blockHashOrNumber: string | number) => Promise<Finding[]>;
  let runTransaction: (txHash: string) => Promise<Finding[]>;
  let logSpy: jest.SpyInstance;

  beforeAll(() => {
    logSpy = jest.spyOn(console, "log");
    logSpy.mockImplementation(() => {});
  });

  beforeEach(async () => {
    const container = configureContainer() as AwilixContainer;
    container.register({
      agentPath: asFunction(
        provideAgentPath(
          "subagents/node-operators-registry/agent-node-operators-registry",
        ),
      ),
      runTransaction: asFunction(provideRunTransaction),
      runBlock: asFunction(provideRunBlock),
    });

    // https://docs.forta.network/en/latest/cli/#invoke-commands-programmatically
    runTransaction = container.resolve("runTransaction");
    runBlock = container.resolve("runBlock");
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it(
    "should process tx with removed signing keys",
    async () => {
      const findings = await runTransaction(
        "0x43529ea1c44b6f6134fa045f6af81a1a15be6309a32424ea742fa056977c4010",
      );
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should process tx with set NO target limit set",
    async () => {
      const findings = await runTransaction(
        "0xcd406d8439cf7b635ede687ea4fbe6d3e3a7d33e1a16c78ca5ba304ac06cb415",
      );
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should process tx with added Node operator",
    async () => {
      const findings = await runTransaction(
        "0xa4629245311d93a11cedb9143d8b7530057685b4b568a026bac194e162002c13",
      );
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );
});
