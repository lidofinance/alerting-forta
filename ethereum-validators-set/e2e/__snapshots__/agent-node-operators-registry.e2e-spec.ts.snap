// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agent-node-operators-registry e2e tests should process tx with added Node operator 1`] = `
Array [
  Finding {
    "addresses": Array [],
    "alertId": "NODE-OPERATOR-ADDED",
    "description": "Node operator 30 added
Name: Launchnodes
Reward address: [0x5a8B929EDBf3CE44526465DD2087EC7EFB59A561](https://etherscan.io/address/0x5a8B929EDBf3CE44526465DD2087EC7EFB59A561)
StakingLimit: 0",
    "labels": Array [],
    "metadata": Object {
      "args": "30,Launchnodes,0x5a8B929EDBf3CE44526465DD2087EC7EFB59A561,0",
    },
    "name": "‚ÑπÔ∏è NO Registry: Node operator added",
    "protocol": "ethereum",
    "severity": 1,
    "type": 4,
  },
  Finding {
    "addresses": Array [],
    "alertId": "NODE-OPERATOR-ADDED",
    "description": "Node operator 31 added
Name: SenseiNode
Reward address: [0xE556Da28015c04F35A52B3111B9F4120E908056e](https://etherscan.io/address/0xE556Da28015c04F35A52B3111B9F4120E908056e)
StakingLimit: 0",
    "labels": Array [],
    "metadata": Object {
      "args": "31,SenseiNode,0xE556Da28015c04F35A52B3111B9F4120E908056e,0",
    },
    "name": "‚ÑπÔ∏è NO Registry: Node operator added",
    "protocol": "ethereum",
    "severity": 1,
    "type": 4,
  },
]
`;

exports[`agent-node-operators-registry e2e tests should process tx with removed signing keys 1`] = `
Array [
  Finding {
    "addresses": Array [],
    "alertId": "NODE-OPERATORS-KEYS-REMOVED",
    "description": "Signing keys has been removed for:
 Kiln: 50",
    "labels": Array [],
    "metadata": Object {},
    "name": "üö® NO Registry: Signing keys removed",
    "protocol": "ethereum",
    "severity": 4,
    "type": 4,
  },
]
`;
